### ISSUE TITLE
Could not load file or assembly Microsoft.Extensions.DependencyInjection.Abstractions, Version=5.0.0.0

### ISSUE THREAD

**Issue by hkzenetex** [dotnet/extensions#3694]
In my azure function app project, I receive an assembly not found runtime error Microsoft.Extensions.DependencyInjection.Abstractions, Version=5.0.0.0. However, the assembly is included in multiple packages I installed from NuGet.

Visual Studio: 2019
Target Framework: netcoreapp3.1
Azure Functions Version: v3

Installed packages containing Microsoft.Extensions.DependencyInjection.Abstractions (5.0.0):
Microsoft.Extensions.Http (5.0.0)
Microsoft.Extensions.Logging (5.0.0)
Microsoft.Extensions.Logging.EventSource (5.0.0)

Error:
A host error has occurred during startup operation...
Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

Ineffective fix attempts:
Installed Microsoft.Extensions.DependencyInjection.Abstractions (5.0.0) from Nuget, but error persists.

Following a similar issue, [#2931](https://github.com/dotnet/extensions/issues/2931#issuecomment-615493815), and [StackOverflow question](https://stackoverflow.com/questions/43995432/could-not-load-file-or-assembly-microsoft-extensions-dependencyinjection-abstrac), I added to the project file the following.

`<PropertyGroup>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
	<GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
  </PropertyGroup>
`

I also followed an article [suggestion](https://azurelessons.com/could-not-load-file-or-assembly-microsoft-extensions-dependencyinjection-abstractions/). After installing Microsoft.Azure.Functions.Extensions, the error persists.

I also followed another [suggestion](https://stackoverflow.com/questions/62434453/azure-functions-host-error-has-occurred-during-startup-operation-could-not-load), and I added the following to my project file. However, the error persists.

`	<runtime>
	  <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
	    <dependentAssembly>
		  <assemblyIdentity name="Microsoft.Extensions.DependencyInjection.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
		    <bindingRedirect oldVersion="5.0.0.0" newVersion="5.0.0" />
		</dependentAssembly>
	  </assemblyBinding>
	</runtime>
`

_20201120_
I attempted the workaround suggested in issue [#401](https://github.com/aspnet/Announcements/issues/401) of ASP.NET Core Announcements: reference the package for the assembly which is failing to load explicitly in my application. However, the error remains.

Using ILSpy, I found Microsoft.Azure.Functions.Extensions, which I installed from NuGet, references Microsoft.Extensions.DependencyInjection.Abstractions, Version 2.1.0.0.

_20201123_
I found a related [StackOverflow question](https://stackoverflow.com/questions/63972495/azure-function-v3-could-not-load-file-or-assembly-microsoft-extensions-dependenc), which links to issue [Azure/azure-functions-vs-build-sdk #472](https://github.com/Azure/azure-functions-vs-build-sdk/issues/472) and finally to issue [Azure/azure-functions-host #6893](https://github.com/Azure/azure-functions-host/issues/6893#issuecomment-726386105). @AartBluestoke summarized: "this can only be solved by upgrading azure functions to a newer version which has a dependency on the 5.x line of the abstractions."

**Comment by olakusibe** [comment]
I am having similar issues, ihave explored same possible solutions and I am still stucked.

**Comment by BrennanConroy** [comment]
Dupe of https://github.com/Azure/azure-functions-host/issues/6893

TLDR; The functions host runs on ~3.1 and will host your app in-process dynamically. What this means is that if the functions host uses 3.1 of DependencyInjection.Abstractions, it will not be able to load 5.0 of the dll in-process for your app, and your app will fail to run.

**Comment by hkzenetex** [comment]
To run my Azure Function project locally, I resolved missing assembly Microsoft.Extensions.DependencyInjection.Abstractions, Version=5.0.0.0 runtime error by referencing the old versions of all its dependent DLLs and by including multiple old versions of this assembly. However, I received errors from GitHub Workflow CI/CD when attempting deployment to Azure.

> NU1605: Detected package downgrade

To deploy my Azure Function App, I resolved the NU1605 error by including in the project file old versions of packages dependent on a **common version** of the original missing assembly, Microsoft.Extensions.DependencyInjection.Abstractions, and I specifically included the common older versions, 3.1.7 and 3.1.2.

  `<ItemGroup>
...
	
	
	
	
	
	
		  

	<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.7" />
	<PackageReference Include="Microsoft.Extensions.Http" Version="3.1.2" />	  
    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.7" />
    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.7" />
    <PackageReference Include="Microsoft.Extensions.Logging.EventSource" Version="3.1.7" />
	<PackageReference Include="Microsoft.Extensions.Options" Version="3.1.7" />	  
    <PackageReference Include="Microsoft.Extensions.Options.DataAnnotations" Version="3.1.2" />
...
  </ItemGroup>
  <ItemGroup>
    <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.2">   
 <HintPath>..\..\..\..\.nuget\packages\microsoft.extensions.dependencyinjection.abstractions\3.1.2\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
    </Reference>
  </ItemGroup>
  <ItemGroup>`
